internal/containers/provider/docker.go

```diff
@@ -42,6 +42,9 @@ type DockerProvider struct {
 type DockerRequest struct {
 	ContainerRequest
 	Mounts []mount.Mount
+	// ManagedVolumes is the list of volumes that should be torn down when the
+	// provider finishes execution
+	ManagedVolumes []mount.Mount
```

Comment:
is the intent to have an `UnmanagedVolumes` sometime in the future?

Reply:
It's not but I couldn't come up with a better name 😅 open to suggestions!

Reply:
Why not just `Volumes`?

Reply:
It could be just `Volumes` I suppose. I wanted to convey the idea that these are managed by the provider though (i.e. the provider is in charge of tearing them down).

Reply:
james is getting at where I was going, but its ultimately a nit so feel free to do nothing!

Reply:
On second thought it might be useful to allow the usage of already existing volumes in the future.